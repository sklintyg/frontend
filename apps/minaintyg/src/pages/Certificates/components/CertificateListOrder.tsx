import { classNames } from '@frontend/components'
import { ReactEventHandler, ReactNode } from 'react'
import { SortDirection } from 'react-stately'

export function SortingButton({ active, onClick, children }: { active: boolean; onClick: ReactEventHandler; children: ReactNode }) {
  return (
    <button
      type="button"
      className={classNames(active ? 'font-semibold pointer-events-none' : 'text-sky-base underline')}
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export function CertificateListOrder({ order, setOrder }: { order: SortDirection; setOrder: (order: SortDirection) => void }) {
  return (
    <div className="text-stone-dark mb-5 flex flex-col md:flex-row md:gap-1">
      Sortera efter:{' '}
      <div className="inline-flex gap-1">
        <SortingButton active={order === 'descending'} onClick={() => setOrder('descending')}>
          Nyast först
        </SortingButton>
        <div role="separator">|</div>
        <SortingButton active={order === 'ascending'} onClick={() => setOrder('ascending')}>
          Äldst först
        </SortingButton>
      </div>
    </div>
  )
}
