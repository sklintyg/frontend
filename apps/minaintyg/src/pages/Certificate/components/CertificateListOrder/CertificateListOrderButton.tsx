import { classNames } from '@frontend/components'
import { ReactEventHandler } from 'react'

export function CertificateListOrderButton({ active, label, onClick }: { active: boolean; label: string; onClick: ReactEventHandler }) {
  return (
    <button
      type="button"
      className={classNames(active ? 'font-semibold pointer-events-none' : 'text-sky-base underline')}
      onClick={onClick}
      aria-label={label}
    >
      {label}
    </button>
  )
}
