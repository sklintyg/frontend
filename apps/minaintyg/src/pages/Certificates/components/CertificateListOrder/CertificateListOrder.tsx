import { SortDirection } from 'react-stately'
import { CertificateListOrderButton } from './CertificateListOrderButton'

export function CertificateListOrder({ order, setOrder }: { order: SortDirection; setOrder: (order: SortDirection) => void }) {
  return (
    <div className="text-stone-dark mb-5 flex flex-col md:flex-row md:gap-1">
      Sortera efter:{' '}
      <div className="inline-flex gap-1">
        <CertificateListOrderButton active={order === 'descending'} label="Nyast först" onClick={() => setOrder('descending')} />
        <div role="separator">|</div>
        <CertificateListOrderButton active={order === 'ascending'} label="Äldst först" onClick={() => setOrder('ascending')} />
      </div>
    </div>
  )
}
