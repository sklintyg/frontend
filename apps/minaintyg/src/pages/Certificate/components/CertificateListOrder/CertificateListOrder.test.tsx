import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { vi } from 'vitest'
import { CertificateListOrder } from './CertificateListOrder'

it('Should render with order set to "descending"', () => {
  const { baseElement } = render(<CertificateListOrder order="descending" setOrder={vi.fn()} />)
  expect(baseElement).toMatchSnapshot()
})

it('Should render with order set to "ascending"', () => {
  const { baseElement } = render(<CertificateListOrder order="ascending" setOrder={vi.fn()} />)
  expect(baseElement).toMatchSnapshot()
})

it('Should return "descending" when newest is pressed', async () => {
  const setOrder = vi.fn()
  render(<CertificateListOrder order="ascending" setOrder={setOrder} />)
  await userEvent.click(screen.getByRole('button', { name: 'Nyast först' }))
  expect(setOrder).toHaveBeenCalledTimes(1)
  expect(setOrder).toHaveBeenCalledWith('descending')
})

it('Should return "ascending" when oldest is pressed', async () => {
  const setOrder = vi.fn()
  render(<CertificateListOrder order="descending" setOrder={setOrder} />)
  await userEvent.click(screen.getByRole('button', { name: 'Äldst först' }))
  expect(setOrder).toHaveBeenCalledTimes(1)
  expect(setOrder).toHaveBeenCalledWith('ascending')
})
