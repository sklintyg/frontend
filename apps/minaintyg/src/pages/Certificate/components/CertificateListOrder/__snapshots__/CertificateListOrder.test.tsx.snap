// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render with order set to "ascending" 1`] = `
<div>
  <div
    class="mb-5 flex flex-col text-stone-dark md:flex-row md:gap-1"
  >
    Sortera efter:
     
    <div
      class="inline-flex gap-1"
    >
      <button
        aria-label="Nyast först"
        class="text-sky-base underline hover:text-sky-dark"
        type="button"
      >
        Nyast först
      </button>
      <div
        role="separator"
      >
        |
      </div>
      <button
        aria-label="Äldst först"
        class="font-semibold pointer-events-none"
        type="button"
      >
        Äldst först
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render with order set to "descending" 1`] = `
<div>
  <div
    class="mb-5 flex flex-col text-stone-dark md:flex-row md:gap-1"
  >
    Sortera efter:
     
    <div
      class="inline-flex gap-1"
    >
      <button
        aria-label="Nyast först"
        class="font-semibold pointer-events-none"
        type="button"
      >
        Nyast först
      </button>
      <div
        role="separator"
      >
        |
      </div>
      <button
        aria-label="Äldst först"
        class="text-sky-base underline hover:text-sky-dark"
        type="button"
      >
        Äldst först
      </button>
    </div>
  </div>
</div>
`;
